### PR TITLE
Fix enum array error

### LINKED ISSUE #46: Arrays of enums don't work
When trying to run squirrel on a query with an array of enums, the following error appears:

```
Error: Query with invalid enum
One of the values in this query is the `my_enum` enum, but I cannot turn it into a Gleam type definition because it has no variants.
```

The issue seems to be here on line 676, where the enum type is resolved with the array OID instead of the enum OID:
https://github.com/giacomocavalieri/squirrel/blob/0ad93fe522378acc0fe9005ba23930d554d98d12/src/squirrel/internal/database/postgres.gleam#L675-L678

### PR DESCRIPTION
**Pull request by leah-u**
This fixes #46. I'm not very experienced with database stuff so I'm happy for any feedback.

The codebase was very pleasant to work with btw :)

### CODE REVIEW

**Comment by giacomocavalieri** [comment]
Thank you so much!! I'm really happy you enjoyed working on this 💜
Right now I'm on a trip without great internet connection but I'll check this as soon as I can!

File: src/squirrel/internal/query.gleam [hunk 1]
```diff
@@ -65,6 +65,39 @@ pub fn add_types(
 ) -> Result(TypedQuery, Error) {
   let UntypedQuery(file:, name:, comment:, content:, starting_line:) = query
 
+  // Quick and dirty hack to allow nullable parameters in insert queries
+  // Looks for a comment like this: '-- nullable: $1, $3' in the sql file
+  // and makes the specified parameters nullable
```
Comment (giacomocavalieri) [inline src/squirrel/internal/query.gleam:70]: Could you please remove this code from the PR? I like the idea and definitely want to add this in the future, but let's keep this PR focused on just the fix for now :)

**Review by giacomocavalieri — CHANGES_REQUESTED** [review]
This looks great, I've just left a comment inline. 
Also could you update the CHANGELOG with the fix?

File: src/squirrel/internal/query.gleam (hunk 1 above)
Comment (leah-u) [reply src/squirrel/internal/query.gleam]: Oops, that wasn't supposed to be there 😅

**Comment by leah-u** [comment]
Should be all good now :)

**Comment by giacomocavalieri** [comment]
Lovely work, thank you! Sorry it took me so long to merge this I had a couple of packed days at a conference 😅

**Pull request merged**